share/spice/currency/currency.js

```diff
@@ -29,29 +222,150 @@
         "xbt": 8
     };
     
-    // Resize the size of the outer container if the content of the inner container
-    // overflows.
-    function resize() {        
-        var resultHeight = $(".zci--currency-result").outerHeight();
-        
-        if(resultHeight > 65) {
-            $(".zci--currency-container").css("height", "9em");
-        } else {
-            $(".zci--currency-container").css("height", "5em");
-        }
-    }
-    
-    // Change the look of the mobile view if the content overflows.
-    function resizeMobile() {
-        var tileHeight = $(".zci--currency .tile--s").outerHeight();
 
-        if(tileHeight > 155) {
-            $(".zci--currency .tile--s").addClass("large").removeClass("small");
-        } else {
-            $(".zci--currency .tile--s").addClass("small").removeClass("large");
-        }
-    }
-    
+    // Currency Converter
+    var Converter = {
+
+        from_currency: undefined,
+        to_currency: undefined,
+        rate: undefined,
+        inverseRate: undefined,
+
+        //
+        // Retrieves conversion info from JSONP payload
+        //
+
+        getConversionRate: function(payload) {
+            return +$(payload["conversion-rate"].split(" ")).get(-2);
+        },
+
+        getInverseConversionRate: function(payload) {
+            return +$(payload["conversion-inverse"].split(" ")).get(-2);
+        },
+
+        getFromCurrencyName: function(payload) {
+            return payload["from-currency-symbol"];
+        },
+
+        getToCurrencyName: function(payload) {
+            return payload["to-currency-symbol"];
+        },
+
+        getConversionLabel: function(payload) {
+            return payload["conversion-rate"];
+        },
+
+        getInverseConversionLabel: function(payload) {
+            return payload["conversion-inverse"]
+        },
+
+        //
+        // Calculates the rates
+        //
+
+        calculateRate: function() {
+            if($currency_input_left.val() !== '' && /\d+/.test($currency_input_left.val())) {
```

Comment:
Minor, but I wonder if we should allow commas in the input. At the moment it just breaks it:
![ddgtemp](https://cloud.githubusercontent.com/assets/4411471/26224822/25fe2608-3bf2-11e7-8e30-bed44428f709.gif)
What if we did a proper format as the user types? i.e. ignore a comma input but automatically insert the commas. (I know this kind of opens up a can of worms about comma vs. decimal formatting in other locales though)